### PR TITLE
🐛 Fix ClusterToKubeadmConfigs to include MachinePools

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
Co-Authored-By: Ace Eldeib <alexeldeib@gmail.com>

**What this PR does / why we need it**: Currently, KubeadmConfigs that belong to a MachinePool do not get a reconciliation request when Cluster Infrastructure becomes ready because the `ClusterToKubeadmConfigs` func only looks for `Machines`. This PR also gets `MachinePool` kubeadmConfigs so the watch on cluster will requeue the KubeadmConfig controller when the cluster is ready.

/cc @alexeldeib 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/656

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
/hold 

I am testing this manually since there are no machine pool e2e tests yet

**Comment by CecileRobertMichon** [comment]
Something's not quite right, will mark as WIP for now while I debug

/retitle [WIP] 🐛 Fix ClusterToKubeadmConfigs to include MachinePools

File: bootstrap/kubeadm/controllers/kubeadmconfig_controller.go [hunk 1]
```diff
@@ -646,6 +646,20 @@ func (r *KubeadmConfigReconciler) ClusterToKubeadmConfigs(o handler.MapObject) [
 		}
 	}
 
+	machinePoolList := &expv1.MachinePoolList{}
```
Comment (alexeldeib) [inline bootstrap/kubeadm/controllers/kubeadmconfig_controller.go:649]: feature gates?

Comment (CecileRobertMichon) [reply bootstrap/kubeadm/controllers/kubeadmconfig_controller.go]: thanks, added

**Comment by CecileRobertMichon** [comment]
All good now 🎉 

```
Every 2.0s: kubectl --kubeconfig=./kubeconfig get nodes                                                                                                                Ceciles-MBP: Tue Aug 11 09:50:18 2020

NAME                            STATUS   ROLES    AGE    VERSION
vmss-md-0-control-plane-28kmm   Ready    master   3m8s   v1.18.5
vmss-md-0-control-plane-999cg   Ready    master   57s    v1.18.5
vmss-md-0-md-0-75687            Ready    <none>   85s    v1.18.5
vmss-md-0-md-0-xt5qv            Ready    <none>   82s    v1.18.5
vmss-md-0-mp-0000000            Ready    <none>   72s    v1.18.5
vmss-md-0-mp-0000002            Ready    <none>   75s    v1.18.5
```

**Comment by devigned** [comment]
> TBH I'm starting to wonder of the fact that MachinePools are not generating machine objects is adding special branches in too many places of the codebase, instead of handling this in the machine controller only. Might be this should be reconsidered before graduating this feature...

@juan-lee, I assume this was discussed during the proposal phase. Do you recall any details?

**Comment by detiber** [comment]
> (Though not strictly related to this PR): TBH I'm starting to wonder of the fact that MachinePools are not generating machine objects is adding special branches in too many places of the codebase, instead of handling this in the machine controller only. Might be this should be reconsidered before graduating this feature...

The main reason for MachinePools was to avoid having to try and map back individual Machines to instances that are managed by cloud provider managed scaling group primitive, I'm not sure we want to re-introduce that requirement. I'd also like to avoid having any type of faked Machine resource for the same purposes.

I do think that it would make sense to require exposing helpers for proper handling of MachinePools vs Machines by bootstrap providers and other non-core consumers as a requirement for graduation, though.

**Comment by CecileRobertMichon** [comment]
Not sure why tide is saying "not mergeable"

**Pull request merged**